lib/ash/resource/change/builtins.ex

```diff
@@ -286,6 +286,45 @@ defmodule Ash.Resource.Change.Builtins do
     {Ash.Resource.Change.Select, target: value, ensure?: true}
   end
 
+  @doc """
+  Cascade this resource's destroy action to a related resource's destroy action.
+
+  Adds an after-action hook that explicitly calls destroy on any records related
+  via the named relationship.  It will optimise for bulk destroys where
+  possible.
+
+  > #### Beware database constraints {: .warning}
+  >
+  > Think carefully before using this change with data layers which enforce
+  > referential integrity (ie PostgreSQL and SQLite) and you may need to defer
+  > constraints for the relationship in question.
+  >
+  > See also:
+  >   1. [PostgreSQL's `SET CONSTRAINTS` documentation](https://www.postgresql.org/docs/current/sql-set-constraints.html)
+  >   2. [SQLite's `PRAGMA defer_foreign_keys` documentation](https://www.sqlite.org/pragma.html#pragma_defer_foreign_keys)
```

Comment:
This is why I originally had it set up as a before_action hook, not an after_action hook, to avoid needing to do this - would that help here?

Reply:
It depends on the relationship type. For cascading to a belongs to then you'd want to do this before the action. If cascading to other relationship types then you'd want to do it after. But also, doing this as a before action hook doesn't work for atomics (because we can only know what we *affected* with an atomic, not what *will be* affected).

Reply:
setting the FK as deferrable initially deferred (to make this work) should be doable with AshPostgres, eg: 

```elixir
    references do
      reference :album, deferrable: :initially
    end
```

AshSqlite says it supports the same option, though I have not tested it

Reply:
I totally forgot that we did that. Probably worth bringing up in this explanation There are some performance implications, but TBH I'm not entirely sure if they are enough to care or not.